File: backup_before_industrialisation_20250626/backend/metier_generator/templates/threed/tests_centralises/api/pagination/test_pagination.py
import unittest


# Exemple de fonction de pagination à tester
def paginate(items, page, page_size):
    start = (page - 1) * page_size
    end = start + page_size
    return items[start:end]


class TestPagination(unittest.TestCase):
    def test_pagination_basic(self):
        items = list(range(1, 21))
        self.assertEqual(paginate(items, 1, 5), [1, 2, 3, 4, 5])
        self.assertEqual(paginate(items, 2, 5), [6, 7, 8, 9, 10])
        self.assertEqual(paginate(items, 4, 5), [16, 17, 18, 19, 20])

    def test_pagination_empty(self):
        items = []
        self.assertEqual(paginate(items, 1, 5), [])

    def test_pagination_out_of_range(self):
        items = list(range(1, 11))
        self.assertEqual(paginate(items, 3, 5), [])


if __name__ == "__main__":
    unittest.main()

File: backup_before_industrialisation_20250626/backend/metier_generator/templates/threed/tests_centralises/api/pagination/test_pagination.js
const paginate = (items, page, pageSize) => {
  const start = (page - 1) * pageSize;
  const end = start + pageSize;
  return items.slice(start, end);
};

test('pagination basic', () => {
  const items = Array.from({ length: 20 }, (_, i) => i + 1);
  expect(paginate(items, 1, 5)).toEqual([1, 2, 3, 4, 5]);
  expect(paginate(items, 2, 5)).toEqual([6, 7, 8, 9, 10]);
  expect(paginate(items, 4, 5)).toEqual([16, 17, 18, 19, 20]);
});

test('pagination empty', () => {
  expect(paginate([], 1, 5)).toEqual([]);
});

test('pagination out of range', () => {
  const items = Array.from({ length: 10 }, (_, i) => i + 1);
  expect(paginate(items, 3, 5)).toEqual([]);
});
